components/core/src/clp/streaming_archive/writer/utils.cpp

```diff
@@ -25,14 +25,17 @@ auto split_file(
         Archive& archive_writer
 ) -> void {
     auto const& encoded_file = archive_writer.get_file();
+    auto has_ts_pattern = encoded_file.has_ts_pattern();
     auto orig_file_id = encoded_file.get_orig_file_id();
     auto split_ix = encoded_file.get_split_ix();
     archive_writer.set_file_is_split(true);
     close_file_and_append_to_segment(archive_writer);
 
     archive_writer.create_and_open_file(path_for_compression, group_id, orig_file_id, ++split_ix);
-    // Initialize the file's timestamp pattern to the previous split's pattern
-    archive_writer.change_ts_pattern(last_timestamp_pattern);
+    if (true == has_ts_pattern) {
```

Comment:
```suggestion
    if (has_ts_pattern) {
```

We only require the `false == condition` pattern to avoid `!condition` (since in the past some devs have missed the `!`, lol).